tox.ini

```diff
@@ -33,6 +33,17 @@ commands =
     black .
     isort .
 
+[testenv:test-all]
+deps =
+    pytest
+    pytest-cov
+    -r {toxinidir}/tests/functional/requirements.txt
+allowlist_externals = mkdir
+commands_pre = mkdir -p tests/report
+commands =
+    pytest {toxinidir}/tests \
+    {posargs:-v --cov --cov-report=xml:tests/report/coverage.xml --junit-xml=tests/report/tests.xml}
```

Comment:
I don't think I'm a fan of this new tox target - it's only used in the tics workflow, and the default args are only useful for the tics workflow. In the interests of simplicity, can the new tics workflow work with what we already have in tox?

For example, calling this from the 'test with tox ...' step in tics.yaml:

```
tox -e func -- -v --cov --cov-report=xml:tests/report/coverage.xml --junit-xml=tests/report/tests.xml
```

And for projects that also have a unit test tox target, I'm sure there is coverage merge functionality, so it could look something like:

```
tox -e func -- -v --cov --cov-report=xml:tests/report/coverage-func.xml --junit-xml=tests/report/tests-func.xml
tox -e unit -- -v --cov --cov-report=xml:tests/report/coverage-unit.xml --junit-xml=tests/report/tests-unit.xml
coverage merge
```

For consistency, we can add support for unit tests in tox here, even if it's a noop.

Reply:
Hmm, it's either a noop target or a target that tests everything. We'll discuss it with the team tomorrow. For me, the justification for `test-all` is that it's an nice entry point to run all tests at once (unit, func)

Reply:
Tox makes it clean to run the environments you want to run - eg. `tox -e unit.func`. There also aren't many cases where you'd want to run unit tests + func tests in a single command.

A noop target might feel a bit of a waste here and now, but if we're wanting to have things as consistent as possible across projects, it might be worth it.

Either way, no real strong opinion, just thinking from terms of consistency. :)